tests/features/camel-hazelcast/src/main/java/org/apache/karaf/camel/test/CamelHazelcastRouteSupplier.java

```diff
@@ -0,0 +1,119 @@
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.karaf.camel.test;
+
+import static org.apache.camel.builder.Builder.constant;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.hazelcast.HazelcastConstants;
+import org.apache.camel.component.hazelcast.HazelcastOperation;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteSupplier;
+import org.apache.karaf.camel.itests.CamelRouteSupplier;
+import org.osgi.service.component.annotations.Component;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+
+@Component(name = "karaf-camel-hazelcast-test", immediate = true, service = CamelRouteSupplier.class)
+public class CamelHazelcastRouteSupplier extends AbstractCamelSingleFeatureResultMockBasedRouteSupplier {
+
+
+    @Override
+    public void configure(CamelContext camelContext) {
+        Config hazelcastConfig = new Config();
+        NetworkConfig networkConfig = hazelcastConfig.getNetworkConfig();
+        networkConfig.setPort(Integer.parseInt(System.getProperty("hz.port")));
+
+        HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(hazelcastConfig);
+
+        camelContext.getRegistry().bind("hzInstance", hazelcastInstance);
+    }
+
+    @Override
+    protected boolean consumerEnabled() {
+        //consumers are implemented directly in the configureProducer method here
+        return false;
+    }
+
+    @Override
+    protected void configureProducer(RouteBuilder builder, RouteDefinition producerRoute) {
+
+        configureConsumer(builder.from("hazelcast-list:myList?hazelcastInstance=#hzInstance").log("list received ${body}").setBody(constant("OK")));
+        configureConsumer(builder.from("hazelcast-map:myMap?hazelcastInstance=#hzInstance").log("map received ${body}").setBody(constant("OK")));
+        configureConsumer(builder.from("hazelcast-replicatedmap:myRMap?hazelcastInstance=#hzInstance").log("rmap received ${body}").setBody(constant("OK")));
+        configureConsumer(builder.from("hazelcast-queue:myQueue?hazelcastInstance=#hzInstance").log("queue received ${body}").setBody(constant("OK")));
+        configureConsumer(builder.from("hazelcast-set:mySet?hazelcastInstance=#hzInstance").log("set received ${body}").setBody(constant("OK")));
+        configureConsumer(builder.from("hazelcast-topic:myTopic?hazelcastInstance=#hzInstance").log("topic received ${body}").setBody(constant("OK")));
```

Comment:
Let's set a different output for each endpoints to distinguish them and quickly identify the one who is failing in case of an error